From 2194895b59850f4c241dbf54c16240110604f0f2 Mon Sep 17 00:00:00 2001
From: Lars Schmertmann <Lars.Schmertmann@governikus.de>
Date: Fri, 6 Dec 2019 07:52:05 +0100
Subject: Remove dead code to fix a lint warning

Obsolete SDK_INT Version Check
------------------------------
This check flags version checks that are not necessary,
because the minSdkVersion (or surrounding known API level)
is already at least as high as the version checked for.

The mindSdkVersion in templates/AndroidManifest.xml is 16 for Qt 5.12
and androiddeployqt also ensures that the minSdkVersion is not below 16.

Change-Id: Ifdb6f773dccdb75c35119df833c976b7b76b3356
Reviewed-by: BogDan Vatra <bogdan@kdab.com>
---
 .../qt5/android/bindings/QtActivityLoader.java  | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git x/qtbase/src/android/java/src/org/qtproject/qt5/android/bindings/QtActivityLoader.java y/qtbase/src/android/java/src/org/qtproject/qt5/android/bindings/QtActivityLoader.java
index 759daf4393..fad9588425 100644
--- x/qtbase/src/android/java/src/org/qtproject/qt5/android/bindings/QtActivityLoader.java
+++ y/qtbase/src/android/java/src/org/qtproject/qt5/android/bindings/QtActivityLoader.java
@@ -132,23 +132,6 @@ public class QtActivityLoader extends QtLoader {
             return;
         }
 
-        if (Build.VERSION.SDK_INT < 16) {
-            // fatal error, show the error and quit
-            AlertDialog errorDialog = new AlertDialog.Builder(m_activity).create();
-            if (m_contextInfo.metaData.containsKey("android.app.unsupported_android_version"))
-                errorDialog.setMessage(m_contextInfo.metaData.getString("android.app.unsupported_android_version"));
-            else
-                errorDialog.setMessage("Unsupported Android version.");
-            errorDialog.setButton(m_activity.getResources().getString(android.R.string.ok), new DialogInterface.OnClickListener() {
-                @Override
-                public void onClick(DialogInterface dialog, int which) {
-                    finish();
-                }
-            });
-            errorDialog.show();
-            return;
-        }
-
         try {
             m_activity.setTheme(Class.forName("android.R$style").getDeclaredField(QT_ANDROID_DEFAULT_THEME).getInt(null));
         } catch (Exception e) {
-- 
2.23.0

